test/livebook/intellisense_test.exs

```diff
@@ -3,22 +3,62 @@ defmodule Livebook.IntellisenseTest do
 
   alias Livebook.Intellisense
 
+  @tmp_dir "tmp/test/remote_intellisense"
+
   # Returns intellisense context resulting from evaluating
   # the given block of code in a fresh context.
-  defmacrop eval(do: block) do
+  defmacrop eval(node \\ nil, do: block) do
     quote do
       block = unquote(Macro.escape(block))
       binding = []
       env = Code.env_for_eval([])
       {value, binding, env} = Code.eval_quoted_with_env(block, binding, env)
+      node = unquote(node) || node()
 
       %{
         env: env,
-        map_binding: fn fun -> fun.(binding) end
+        map_binding: fn fun -> fun.(binding) end,
+        node: node
       }
     end
   end
 
+  setup_all do
+    {:ok, _pid, node} =
+      :peer.start(%{
+        name: :remote_runtime,
+        args: [~c"-setcookie", Atom.to_charlist(Node.get_cookie())]
+      })
+
+    {:module, Elixir.RemoteModule, bytecode, _} =
+      defmodule Elixir.RemoteModule do
+        @compile {:autoload, false}
+        @moduledoc """
+        Remote module docs
+        """
+
+        @doc """
+        Hello doc
+        """
+        def hello(message) do
+          message
+        end
+      end
+
+    false = Code.ensure_loaded?(RemoteModule)
+
+    File.rm_rf!(@tmp_dir)
+    File.mkdir_p!(@tmp_dir)
+    File.write!("#{@tmp_dir}/Elixir.RemoteModule.beam", bytecode)
+
+    true = :erpc.call(node, :code, :set_path, [:code.get_path() ++ [~c"#{@tmp_dir}"]])
+    {:ok, _} = :erpc.call(node, :application, :ensure_all_started, [:elixir])
+    {:module, RemoteModule} = :erpc.call(node, :code, :load_file, [:"Elixir.RemoteModule"])
+    :loaded = :erpc.call(node, :code, :module_status, [:"Elixir.RemoteModule"])
```

Comment:
To simplify we could use `Livebook.Runtime.ElixirStandalone`. Start a runtime, evaluate code that defines the module, and then use `runtime.node`.

Reply:
That's a good idea but we can end-up accidentally polluting the node in tests and not find bugs. So I would try to keep the remote node as "clean" as possible, if that makes any sense.

Reply:
All the other tests run in the context of Livebook itself, so there is definitely much more polluted in that sense.

Reply:
What do you mean? To be clear, what I mean is that, if the remote node has Livebook's runtime, then we can accidentally do `:erpc.call(node(), SomeLivebookModule)` which will not be available in actual remote nodes, only in tests.

Reply:
> What do you mean?

That when testing completion all the livebook modules could be listed, we just have the tests such that this is never the case.

> then we can accidentally do :erpc.call(node(), SomeLivebookModule)

I see, we are clearly just fetching docs info so I'm not worried about that, but that's a fair point, we can keep the minimal version.